### PR TITLE
Add `ExecutionWitness` to the `ExecutionPayload` to support the Verge

### PR DESCRIPTION
**Pull request by macladson**
## Proposed Changes

Implement the draft spec for the Verge: https://github.com/ethereum/consensus-specs/pull/3230
This succeeds the more basic implementation used on the first [Kaustinen](https://kaustinen.ethdevops.io/) testnet (see #3639)

## Additional Info

Does not currently contain any fork implementation, instead it adds the new field directly into the Bellatrix structures.
Contains both https://github.com/sigp/lighthouse/pull/3807 (used for local testing) and https://github.com/sigp/lighthouse/pull/3644

## Checklist

- [x] Add `ExecutionWitness` to `ExecutionPayload` and `ExecutionPayloadHeader`.
- [ ] Add placeholder fork implementation
- [ ] Add tests (and fix existing ones)

### CODE REVIEW

File: consensus/types/src/execution_witness.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+#[derive(
+    Default, Debug, Clone, Serialize, Deserialize, Encode, Decode, TreeHash, TestRandom, Derivative,
+)]
+#[derivative(PartialEq, Hash(bound = "T: EthSpec"))]
+#[serde(bound = "T: EthSpec")]
+#[serde(rename_all = "camelCase")]
+pub struct SuffixStateDiff<T: EthSpec> {
+    //#[serde(with = "eth2_serde_utils::quoted_u8")]
```
Comment (paulhauner) [inline consensus/types/src/execution_witness.rs:63]: It's unclear to me if this warrants quotes when serialized as JSON, but it shouldn't be consensus breaking and we can figure that out later.

File: consensus/types/src/execution_witness.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    suffix: u8,
+    // `None` means not currently present.
+    current_value: Option<StateDiffValue<T>>,
```
Comment (paulhauner) [inline consensus/types/src/execution_witness.rs:66]: The Verge spec isn't clear on how this should be serialized, but I believe our implementation uses `Union[None, T]` which I think is the latest version to *almost* make it into the spec 😅 

There is a [contender](https://eips.ethereum.org/EIPS/eip-6475), which is different to what we have, but it's not clear that we should switch to that IMO.

File: consensus/types/src/execution_witness.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+use tree_hash_derive::TreeHash;
+
+#[derive(
+    Default, Debug, Clone, Serialize, Deserialize, Encode, Decode, TreeHash, TestRandom, Derivative,
+)]
+#[derivative(PartialEq, Hash(bound = "T: EthSpec"))]
+#[serde(bound = "T: EthSpec")]
+#[ssz(struct_behaviour = "transparent")]
+#[serde(transparent)]
+pub struct BandersnatchGroupElement<T: EthSpec> {
```
Comment (paulhauner) [inline consensus/types/src/execution_witness.rs:15]: If you're using these transparent structs just for readability, you could also do something like:

```rust
pub type BandersnatchGroupElement<T> = FixedVector<u8, T>;
```

**Review by paulhauner — COMMENTED** [review]
I've had a look through the SSZ encoding of the new structs in `consensus/types` and they look good to me! I've left a couple of notes, nothing special.

Let me know if there are other structs you'd like me to look at ☺️ 

Regarding the transparent SSZ types PR, if you're using #3644 then you should be fine. The version in `unstable` has more features but should encode/decode the same.

**Comment by macladson** [comment]
Closed in favour of #4891

**Pull request closed without merging**